admin/RequestSolrCloudBackups

```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+
+use Getopt::Long qw( GetOptions );
+use Pod::Usage qw( pod2usage );
+
+################################################################################
+
+=head1 NAME
+
+RequestSolrCloudBackups - Request MB SolrCloud leader to backup every collection
+
+=head1 SYNOPSIS
+
+RequestSolrCloudBackups [options]
+
+Options:
+
+    -c, --collection COLLECTION         backup given collection only
+    -d, --debug                         print more progress messages
+
+    -h, --help                          show this help
+
+Environment:
+
+    BACKUP_STAMP                        timestamp used for backup directory name
+                                        (default: date +%Y%m%d-%H%M%S GMT)
+
+=head1 COPYRIGHT AND LICENSE
+
+Copyright (C) 2019 MetaBrainz Foundation
+
+This file is part of MusicBrainz, the open internet music database,
+and is licensed under the GPL version 2, or (at your option) any
+later version: http://www.gnu.org/licenses/gpl-2.0.txt
+
+=cut
+
+################################################################################
+
+my $sample_collection;
+my $debug_flag;
+my $help_flag;
+
+GetOptions(
+    "collection|c=s"            => \$sample_collection,
+    "debug|d"                   => \$debug_flag,
+    "help|h"                    => \$help_flag,
+);
+
+pod2usage() if $help_flag;
+pod2usage(
+    -exitval => 64, # EX_USAGE
+    -message => "$0: unrecognized arguments",
+) if @ARGV;
+
+################################################################################
+
+$SIG{'INT'} = sub { exit 3 };
+
+use JSON::XS qw( decode_json encode_json );
+use List::Util qw( sum );
+use POSIX qw( strftime );
+
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+
+use DBDefs;
+use MusicBrainz::Server::Context;
+
+################################################################################
+
+sub _debug { print "[debug] ", (sprintf shift, @_), "\n" if $debug_flag; }
+sub _info { print "[info] ", (sprintf shift, @_), "\n"; }
+sub _warn { warn "[warning] ", (sprintf shift, @_), "\n"; }
+sub _error { warn "[error] ", (sprintf shift, @_), "\n"; }
+
+foreach my $def (qw(
+    SOLRCLOUD_COLLECTIONS_API
+    SOLRCLOUD_BACKUP_LOCATION
+))
+{
+    if (!defined DBDefs->$def || !DBDefs->$def)
+    {
+        _error("$def is not set in DBDefs.");
+        exit 78; # EX_CONFIG
+    }
+}
+
+my $COLLECTIONS_API = DBDefs->SOLRCLOUD_COLLECTIONS_API;
+my $BACKUP_LOCATION = DBDefs->SOLRCLOUD_BACKUP_LOCATION;
+
+my $BACKUP_STAMP = $ENV{BACKUP_STAMP} // strftime "%Y%m%d-%H%M%S", gmtime;
+
+my $c = MusicBrainz::Server::Context->create_script_context;
+
+#my %BACKUP_REQUEST_STATUS = (
```

Comment:
It was supposed to document the hard-coded values `'pending'`, `'running'`, and `'completed'`, but replacing `'pending'` with `BACKUP_REQUEST_STATUS['pending']` and so on does not seem nice either…

Would using `Readonly` instead be alright?
```perl
Readonly our $BACKUP_REQUEST_PENDING = 'pending'; # Backup request has not been sent yet
```

Reply:
Ah okay, I didn't realize it was documentation (thought it might be unused code). The readonly constants sound fine, or maybe just turning the comment here into prose so it doesn't look like dead code.

Reply:
Used `Readonly` in https://github.com/metabrainz/musicbrainz-server/pull/1407/commits/b5c1b88f54fc81b8cca67e550c548e9576bef171 to better distinguish status in script from status in SolrCloud.